### PR TITLE
Enhanced request id generation

### PR DESCRIPTION
**Pull request by McReader**
https://www.notion.so/Oracle-Uniq-ID-generation-0dafce6ef1dd4cb0a3d6eafed63dae82?pvs=4

This PR Contains Breaking Changes ‼️ IOracle interface got the following methods updated: addRequest, fallbackCall

### CODE REVIEW

File: src/Oracle.sol [hunk 1]
```diff
@@ -43,28 +48,32 @@ contract Oracle is IOracle, DataStreamConsumer, PriceFeedConsumer {
...
         // This contains the data returned by your implementation in checkCallback().
-        (bytes[] memory signedReports, bytes memory extraData) = abi.decode(
-            performData,
-            (bytes[], bytes)
-        );
+        (
+            bytes[] memory signedReports,
+            bytes memory extraData,
+            uint256 nonce,
+            address sender
```
Comment (killroy192) [inline src/Oracle.sol]: nonce and sender will be inside `extraData`

**Review by killroy192 — CHANGES_REQUESTED** [review]

File: src/mocks/MockConsumer.sol [hunk 2]
```diff
@@ -39,7 +43,9 @@ contract MockConsumer is IOracleConsumerContract {
     function trigger(CustomRequestParams memory params) public returns (bool) {
         bool success = IOracle(oracle).addRequest(
             address(this),
-            abi.encode(params)
+            abi.encode(params),
```
Comment (killroy192) [inline src/mocks/MockConsumer.sol:46]: params have a CustomRequestParams interface, and already includes nonce and msg.sender

File: src/mocks/MockOracle.sol [hunk 3]
```diff
@@ -18,18 +23,25 @@ contract MockOracle is IFakeOracle, Oracle {
 
     function addFakeRequest(
         address callbackContract,
-        bytes memory callbackArgs
+        bytes memory callbackArgs,
```
Comment (killroy192) [inline src/mocks/MockOracle.sol:26]: nonce and sender are already in callbackArgs

**Review by killroy192 — CHANGES_REQUESTED** [review]

File: src/Oracle.sol (hunk 1 above)
Comment (McReader) [reply src/Oracle.sol]: Fixed

**Review by killroy192 — APPROVED** [review]

**Pull request merged**